### DISCUSSION TITLE
Logging Power Outages & Spikes

### DISCUSSION THREAD

**Discussion by lordofscripts** [David00/rpi-power-monitor#132] [Q&A]
This is my situation, in my country the service isn't that great. A few years ago a voltage spike not only damaged $1200 of computer equipment but also the voltage protector. It happened a 2nd time with less losses.

So, in my situation it would be nice to monitor a circuit to **document** power outages as well as certain spikes _(brownouts)_ in order to document claims of at least have an idea of the kind of service I am getting at the location.

**Comment by David00** [comment]
Hi @lordofscripts, thanks for kicking off a discussion topic about this.  That is very unfortunate that your grid service has damaged your gear on multiple occasions!

For recording outages, this would most easily be identifiable if the AC voltage input drops to zero (assuming you have the Pi on a battery backup!) Creating a log of events would be relatively easily done with a custom plugin and you could write the events to a text file, CSV file, or even send a HTTP request to another system.  The docs for creating and working with plugins are here:
https://david00.github.io/rpi-power-monitor/docs/v0.3.0/plugins.html

For detecting spikes, the power monitor is not capable as written to detect or report on spikes.  The power monitor readings that are reported/stored are basically multiple ~0.5 second interval readings averaged together, and these averages will essentially mask any short duration voltage spikes.  

The power monitor does indeed sample quick enough to gather the data needed to detect voltage spikes, but the current way that the processing of the data is done cannot detect millisecond or sub-millisecond voltage spikes. Hopefully that makes sense!
